compiler/rustc_target/src/asm/mod.rs

```diff
@@ -298,54 +299,87 @@ impl InlineAsmReg {
         arch: InlineAsmArch,
         target_features: &FxHashSet<Symbol>,
         target: &Target,
+        is_clobber: bool,
```

Comment:
I wonder if it wouldn't make sense to proactively make this a tri-state enum describing the context (clobber, input, output). I can imagine it being useful when we decide to actually start adding constraint classes that are applicable to input or output only.

Reply:
I have a followup PR based on this one which is a pain to rebase. I'll make the change there.